ci/taskcluster/bin/decision.py

```diff
@@ -7,15 +7,23 @@
 
 tasks = [
     {
+        'name': 'recipe-client-addon:build',
+        'description': 'Build Firefox with recipe-client-addon',
+        'command': 'normandy/recipe-client-addon/bin/tc/build.sh',
+    },
+    {
         'name': 'recipe-client-addon:test',
         'description': 'Test recipe-client-addon with gecko-dev',
         'command': 'normandy/recipe-client-addon/bin/tc/test.sh',
+        'dependencies': ['recipe-client-addon:build'],
+        'artifacts_from': {
+            'BUILD_RESULT': {'task': 'recipe-client-addon:build', 'path': 'build.tar.gz'},
+        },
     },
     {
         'name': 'recipe-client-addon:make-xpi',
         'description': 'Build XPI for recipe-client-addon',
         'command': 'normandy/recipe-client-addon/bin/tc/make-xpi.sh',
-        'dependencies': ['recipe-client-addon:test'],
```

Comment:
We don't technically need the tests to pass before making an XPI, and this may serve as an interesting debugging tool. I'd be happy changing this to depend on the build, tests, or both.